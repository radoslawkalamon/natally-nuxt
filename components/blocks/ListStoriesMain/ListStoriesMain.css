.list-stories-main__title {
  text-align: center;
}

.list-stories-main__stories {
  display: grid;
  gap: var(--gap);
  grid-template-columns: repeat(auto-fit, minmax(292px, 1fr));
  justify-items: center;
}
