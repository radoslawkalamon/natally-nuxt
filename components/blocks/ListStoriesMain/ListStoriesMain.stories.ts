import type { Meta, Story } from '@storybook/vue'
import BlocksListStoriesMain from '@/components/blocks/ListStoriesMain/ListStoriesMain.vue'
import { storybookOnlyDevTemplate } from '@/devtools/storybook.onlyDev.template'

const meta: Meta = {
  title: 'Blocks / List Stories Main',
  component: BlocksListStoriesMain
}
export default meta

export const Default: Story = () => ({
  components: {
    BlocksListStoriesMain
  },
  template: process.env.NODE_ENV !== 'production'
    ? '<div class="__storybook-section__"><BlocksListStoriesMain /></div>'
    : storybookOnlyDevTemplate
})
