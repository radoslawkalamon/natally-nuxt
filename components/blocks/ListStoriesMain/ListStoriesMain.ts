import Vue from 'vue'

export default Vue.extend({
  name: 'BlocksListStoriesMain',
  components: {
    ComponentsCardStory: () => import(/* webpackChunkName: "components" */'@/components/components/CardStory/CardStory.vue'),
    ComponentsTitle: () => import(/* webpackChunkName: "components" */'@/components/components/Title/Title.vue'),
    WrappersListStories: () => import(/* webpackChunkName: "wrappers" */'@/components/wrappers/ListStories/ListStories.vue'),
    WrappersSection: () => import(/* webpackChunkName: "wrappers" */'@/components/wrappers/Section/Section.vue')
  }
})
