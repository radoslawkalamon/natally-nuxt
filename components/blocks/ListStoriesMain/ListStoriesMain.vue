<template>
  <WrappersSection
    class="list-stories-main"
    data-test="blocks-list-stories-main"
  >
    <template #header>
      <ComponentsTitle
        class="list-stories-main__title"
        :design="2"
        title="Opowiadania"
        :type="2"
      />
    </template>

    <WrappersListStories class="list-stories-main__stories">
      <template #default="{ metaPostStories }">
        <template v-for="metaPostStory in metaPostStories">
          <ComponentsCardStory
            :key="metaPostStory.path"
            :meta-post-story="metaPostStory"
          />
        </template>
      </template>
    </WrappersListStories>
  </WrappersSection>
</template>

<script src="./ListStoriesMain.ts" lang="ts"></script>
<style src="./ListStoriesMain.css" scoped></style>
