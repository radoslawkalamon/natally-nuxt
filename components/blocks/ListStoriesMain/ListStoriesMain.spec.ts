import { merge } from 'lodash'
import ListStoriesMain from './ListStoriesMain.vue'
import { shallRender } from '@/devtools/jest.shared.spec'

const defaultOptionsFactory = (options?: object) => merge({
  stubs: [
    'WrappersSection',
    'ComponentsTitle',
    'WrappersListStories',
    'ComponentsCardStory'
  ]
}, options)

describe('Blocks / List Stories Main', () => {
  shallRender(ListStoriesMain, defaultOptionsFactory())
})
