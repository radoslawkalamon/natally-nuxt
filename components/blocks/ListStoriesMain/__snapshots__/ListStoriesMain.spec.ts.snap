// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blocks / List Stories Main Integration shall pass integration sanity test 1`] = `
"<section class=\\"section list-stories-main\\">
  <header data-test=\\"wrappers-section-header\\" class=\\"section__header\\">
    <h2 class=\\"list-stories-main__title title title--design-2 title--underscore\\">Opowiadania</h2>
  </header>
  <div class=\\"section__inner\\">
    <div class=\\"list-stories list-stories-main__stories\\"></div>
  </div>
  <!---->
</section>"
`;

exports[`Blocks / List Stories Main Unit shall pass unit sanity test 1`] = `
"<wrapperssection-stub class=\\"list-stories-main\\">
  <wrappersliststories-stub class=\\"list-stories-main__stories\\"></wrappersliststories-stub>
</wrapperssection-stub>"
`;
